pennylane/tape/circuit_graph.py

```diff
@@ -48,6 +53,26 @@ def observables(self):
         """Observables in the circuit."""
         return self._observables
 
+    @property
+    def depth(self):
```

Comment:
I tend to like properties when they need to perform some additional validation outside of a simple attribute, but for something a bit more computational intensive (like a graph algorithm), I tend to favour methods.

I find that properties can 'hide' the computation a bit, whereas with a method you expect it. E.g.,

* `tape.depth`: this feels like a pre-computed quantity I am extracting, and might be surprised if its not instantaneous

* `tape.get_depth()`: I would not be surprised if this takes some time to compute

This is relatively minor, just curious to get peoples thoughts on it

Reply:
I was curious, so I did a really rough timing test for 2 - 10 wires, for some simple random circuits with up to 50000 gates and a single expectation value at the end, so it will compute the truncated graph. Worst case it's like close to 2s. Though this doesn't tell us the whole story, e.g. maybe there are some weird edge case circuits, and these are single circuits at each point and not averages (though the results of a couple runs produced very similar plots). In any case I'm comfortable with `tape.depth`.

![image](https://user-images.githubusercontent.com/2068515/96922769-47532c00-147e-11eb-9d03-c86606211dd5.png)

Reply:
I  would say that 2s compute time would definitely convince me for `get_depth()` 😆 But then again, that is for 50000 gates, which isn't a common use case. Happy with either for now, we can always iterate with feedback.

Reply:
After doing some research and querying my in-house software dev oracle, I think this should be switched to `get_resources()` as well as `get_depth()` for both `TapeCircuitGraph` and `QuantumTape`. If they were things that would be computed upon instantiation, `@property` would make sense because that content is already there and just being accessed. However for depth and resources, as much as I consider them to be properties of a circuit, here they aren't computed until asked for. Furthermore they will change if you add/remove stuff from the tape, and require recomputation.